app/models/subject_selector.rb

```diff
@@ -0,0 +1,25 @@
+class SubjectSelector
+  def self.for(workflow)
+    case strategy_for(workflow)
```

Comment:
could we push responsibility to the workflow to hide the implementation? perhaps
``` ruby
case 
when workflow.strategy?(:cellect)
...
```